### ISSUE TITLE
Write/read mismatches (found by static analysis)

### ISSUE THREAD

**Issue by amaembo** [cdk/cdk#531]
More issues found by IntelliJ IDEA static analyzer. This time: mismatched writes and reads of mutable objects.

1. org.openscience.cdk.interfaces.AbstractPolymerTest#testToString
```java
Map<String, IMonomer> monomers = new Hashtable<String, IMonomer>();
monomers.put("TRP279", oMono1);
monomers.put("HOH", oMono2);
```
This Map is populated, but not used afterwards. Not sure, probably it should be added somehow to `polymer` or used in assertions or just removed.

Similar - org.openscience.cdk.interfaces.AbstractStrandTest#testToString
2. org.openscience.cdk.tools.manipulator.AtomContainerManipulator#suppressHydrogens
line 767: `Set<IBond> bondsToHydrogens = new HashSet<IBond>();` is declared, line 992: it's iterated via `for (IBond bondToHydrogen : bondsToHydrogens)`, but it never filled. Probably filling code is missing, or the iteration could be removed.
3. org.openscience.cdk.renderer.visitor.AWTDrawVisitor#map
map field (line 109) is filled only once (line 156) and never queried. Probably could be removed.
4. org.openscience.cdk.ringsearch.FiguerasSSSRFinder#getRing
`path` list (line 197) is updated (line 201), but never queried. Probably could be removed.
5. org.openscience.cdk.inchi.InChIGenerator#generateInchiFromCDKAtomContainer
`bondMap` (line 273) is updated (line 299), but never queried. Not sure whether it's unnecessary or query code is missing.
6. org.openscience.cdk.qsar.descriptors.atomic.IPAtomicHOSEDescriptor.IPdb
Both fields `listGroup` and `listGroupS` are queried (lines 229-231), but never written. Probably this code is incomplete or populating constructor is missing or these fields could be removed.
Similar - org.openscience.cdk.qsar.descriptors.atomic.ProtonAffinityHOSEDescriptor.Affinitydb
7. org.openscience.cdk.graph.PathTools#getShortestPath
`sList` (line 463) is updated (line 481), but never queried. Probably unnecessary.
8. org.openscience.cdk.reaction.type.PiBondingMovementReactionTest#testAutomaticSearchCentreActiveExample1
`paramList` (line 167) is updated (line 170), but never queried
org.openscience.cdk.reaction.type.PiBondingMovementReactionTest#testDoubleRingConjugated - similar
9. org.openscience.cdk.io.PMPReader#atomZOrders
`atomZOrders` field (line 86) is updated (line 244), but never queried. Probably reserved for future use or may be removed.
10. org.openscience.cdk.charges.Polarizability#calculateGHEffectiveAtomPolarizability(org.openscience.cdk.interfaces.IAtomContainer, org.openscience.cdk.interfaces.IAtom, boolean, int[][])
`startAtom` list (line 170) is updated (line 171), but never queried.
11. org.openscience.cdk.fingerprint.ShortestPathWalker#pseudoAtoms
`pseudoAtoms` list (line 61) is updated (line 160), but never queried.
12. org.openscience.cdk.ringsearch.cyclebasis.SimpleCycleBasis.AuxiliaryGraph2#auxEdgeMap
`auxEdgeMap` field (line 872) is updated many times, but never queried.
13. org.openscience.cdk.qsar.descriptors.molecular.BCUTDescriptor#calculate
`names` array is created and populated (lines 474-483), but never used afterwards, so this loop essentially does nothing.
14. org.openscience.cdk.smiles.DeduceBondSystemTool#sixMemberedRingPossibilities
`ringatoms` array (line 337) first element is written (line 339), and never used afterwards. Probably could be removed.
org.openscience.cdk.smiles.DeduceBondSystemTool#sevenMemberedRingPossibilities - similar
15. org.openscience.cdk.fingerprint.IntArrayFingerprint#set(int, boolean)
`tmp` array is written (lines 169-171), but not used afterwards. Probably `trueBits = tmp;` assignment is missing (it's performed in another `if` branch).
16. org.openscience.cdk.qsar.descriptors.molecular.SmallRingDescriptor#atomArom
`atomArom` field (line 71) is written (lines 422, 472, 562), but never read. Probably reserved for future use or may be removed.

**Comment by johnmay** [comment]
At first class some of these are deprecated/legacy code. Please feel free to add a pull request.

**Comment by johnmay** [comment]
Fixed all except

4. - deprecated won't fix
12. - deprecated won't fix
14. - deprecated won't fix

also

9. - @egonw is original author, cleaned up class and remove ZOrder as other parts of file also seem to be ignored

**Comment by johnmay** [comment]
resolved
